bears/java/PMDBear.py

```diff
@@ -0,0 +1,35 @@
+import re
+
+from coalib.bearlib.abstractions.Lint import Lint
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+
+
+class PMDBear(LocalBear, Lint):
+    executable = 'run.sh'
+    arguments = "pmd -R java-basic," + \
+                "java-braces," + \
+                "java-clone," + \
+                "java-codesize," + \
+                "java-controversial," + \
+                "java-design," + \
+                "java-imports," + \
+                "java-naming," + \
+                "java-optimizations," + \
+                "java-strings," + \
+                "java-unnecessary," + \
+                "java-unusedcode" + \
+                " -d {filename}"
+    output_regex = re.compile(
+        r'(?P<file_name>.+):'
+        r'(?P<line>.+):'
+        r'(?P<message>.*)')
+    severity_map = {
+        "error": RESULT_SEVERITY.MAJOR,
+        "warning": RESULT_SEVERITY.NORMAL}
+
+    def run(self, filename, file):
+        '''
+        Checks the code with ``pmd``.
```

Comment:
As far as I have researched, pmd cannot do much more. Java has the most rulesets. Other supported languages(Javascript, JSP, Velocity, PL/SQL, XML) have very basic rulesets that are most probably covered in the existing bears, which offer much more. What is technically supported on all languages is CPD(code duplicate) which could represent a new, separate bear :D